pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,323 @@
+					defer mutex.Unlock()
+					if shouldBindPspt {
+						shouldBindPspt = false
+
+						google11, err := m.clusterLister.Get("", "local")
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+						google11.Spec.DefaultPodSecurityPolicyTemplateName = "pspt1000"
+
+						_, err = m.clusters.Update(google11)
+
+						if err != nil {
+							logrus.Errorf("%v", err)
+							os.Exit(1)
+						}
+
+					}
+				}
+
+				if true {
+					r()
+				}
+			*/
+			logrus.Info("handle service account")
+
+			if obj == nil {
+				logrus.Infof("no service account provided, exiting")
+				// do nothing
+				return nil
+			}
+
+			namespaceName := obj.Namespace
+
+			logrus.Infof("svc acct name: %v namespace: %v", obj.Name, namespaceName)
+
+			// Check if a PSP exists for this cluster
+			policies, err := m.policyLister.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting policies: %v", err)
+			}
+
+			bindings, err := m.bindingLister.List(namespaceName, labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting role bindings: %v", err)
+			}
+
+			// get PSPT
+			// if no PSPT then get default
+			// if no default then exit
+			namespace, err := m.namespaceLister.Get("", namespaceName)
+
+			if err != nil {
+				return fmt.Errorf("error getting namespaces: %v", err)
+			}
+
+			annotation := namespace.Annotations[projectIdAnnotation]
+
+			if annotation == "" {
+				// no project is associated with namespace so don't do anything
+				logrus.Infof("no project is associated with namespace %v", namespaceName)
+				return nil
+			}
+
+			split := strings.Split(annotation, ":")
+
+			if len(split) != 2 {
+				return fmt.Errorf("could not parse project id annotation: %v", annotation)
+			}
+
+			clusterName, projectId := split[0], split[1]
+
+			projects, err := m.projectLister.List("", labels.Everything())
+
+			var project *v3.Project
+
+			for _, candidate := range projects {
+				if candidate.Name == projectId {
+					project = candidate
+					break
+				}
+			}
+
+			if err != nil {
+				return fmt.Errorf("error getting projects: %v", err)
+			}
+
+			podSecurityPolicyTemplateId := project.Spec.PodSecurityPolicyTemplateName
+
+			if podSecurityPolicyTemplateId == "" {
+				// check cluster
+				cluster, err := m.clusterLister.Get("", clusterName)
+
+				if err != nil {
+					return fmt.Errorf("error getting clusters: %v", err)
+				}
+
+				if cluster == nil {
+					return fmt.Errorf("could not find cluster %v", clusterName)
+				}
+
+				podSecurityPolicyTemplateId = cluster.Spec.DefaultPodSecurityPolicyTemplateName
+
+				if podSecurityPolicyTemplateId == "" {
+					logrus.Infof("No PSPTs found for project %v and cluster %v", projectId, clusterName)
+					return nil
+				}
+			}
+
+			var policy *v1beta12.PodSecurityPolicy
+
+			for _, candidate := range policies {
+				if candidate.Annotations[podSecurityAnnotation] == podSecurityPolicyTemplateId {
```

Comment:
Seems like this should be a label instead of an annotation and then you should do the list call with a proper label selector.
also, error out if more than one is returned